plugins/obs-ffmpeg/obs-ffmpeg-vaapi.c

```diff
@@ -1074,6 +1074,20 @@ static bool get_device_name_from_pci(struct pci_access *pacc, char *pci_slot,
 	return false;
 }
 
+static bool vaapi_device_codec_supported(const char *path,
+					 enum codec_type codec)
+{
+	switch (codec) {
+	case CODEC_H264:
+		return vaapi_device_h264_supported(path);
+	case CODEC_HEVC:
+		return vaapi_device_hevc_supported(path);
+	case CODEC_AV1:
+		return vaapi_device_av1_supported(path);
+	}
+	return false;
```

Comment:
Nit:

```suggestion
	default:
		return false;
	}
```

Reply:
Actually i want it outside the switch to keep all the branches covered. With default we wont hit the warning if someone adds a new codec (unlikely, but still).

Reply:
Normally we should remove `-Wno-error=switch`, but QSV…